library/core/src/time.rs

```diff
@@ -1043,12 +1043,18 @@ impl fmt::Debug for Duration {
         /// to the formatter's `width`, if specified.
         fn fmt_decimal(
             f: &mut fmt::Formatter<'_>,
-            mut integer_part: u64,
+            integer_part: u64,
             mut fractional_part: u32,
             mut divisor: u32,
             prefix: &str,
             postfix: &str,
         ) -> fmt::Result {
+            // If integer_part == u64::MAX and precision < 9, any carry of the
+            // overflow during rounding of the fractional_part into the
+            // integer_part will cause the integer_part itself to overflow.
+            // Avoid this by using a u128.
+            let mut integer_part: u128 = integer_part.into();
```

Comment:
Pondering: it's a bit of a shame to force this to deal in `u128`, which generally isn't handled as well on extant processors, when it only needs to handle one more value.  (It only needs 0.00000000000000000008 more bits.)

How bad would it be to replace
```rust
integer_part += 1;
```
below with something like
```rust
(integer_part, overflowed) = integer_part.overflowing_add(1);
```
instead and deal with the `overflowed` case separately when printing?  After all, if it can only overflow to one possible value we could just hard-code the display for that one, rather than doing things like `ilog10` on 128-bit numbers.

If that gets kinda messy, though, then feel free to describe how and we can leave that to a future PR.